Src/AutoMoq/AutoMockPropertiesCommand.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using Moq;
+using Ploeh.AutoFixture.Kernel;
+
+namespace Ploeh.AutoFixture.AutoMoq
+{
+    /// <summary>
+    /// A command that populates all public writable properties/fields of a mock object with anonymous values.
+    /// </summary>
+    public class AutoMockPropertiesCommand : ISpecimenCommand
```

Comment:
You're completely right though. The two snippets should return the same object.

This means any post-processing needs to be done on the instance of `Mock<IHasProperties>` returned by the customization (which is executed in both cases), and not on the instance of `IHasProperties` returned by the residue collector (which is executed only in the second case).

As follows, it seems like we do need this `AutoMockPropertiesCommand` to unwrap `Mock<IHasProperties>.Object` and pass it to an `AutoPropertiesCommand` after all, even though the initial reasoning for doing this was flawed.